src/py/_pyodide/_base.py

```diff
@@ -649,8 +650,11 @@ def find_imports(source: str) -> list[str]:
 
 
 def pyimport_impl(path: str) -> Any:
-    [stem, *fromlist] = path.rsplit(".", 1)
-    res = __import__(stem, fromlist=fromlist)
-    if fromlist:
-        res = getattr(res, fromlist[0])
-    return res
+    if "." not in path:
+        return import_module(path)
+    stem, name = path.rsplit(".", 1)
+    mod = import_module(stem)
+    try:
+        return getattr(mod, name)
+    except AttributeError:
+        return import_module(path)
```

Comment:
How about this:
```suggestion
    # If multiple components, behave like `from a.b.c import last_part`
    # This compiles into 
    # IMPORT_NAME a.b.c
    # IMPORT_FROM last_part
    # for only one component, we want to behave like IMPORT_NAME
    [stem, *fromlist] = path.rsplit(".", 1)
    res = __import__(stem, fromlist=fromlist)
    if not fromlist:
       # Only one component, no IMPORT_FROM step so we can return now
       return res
    # Now imitate IMPORT_FROM
    try:
        return getattr(mod, name)
    except AttributeError:
        return import_module(path)
```

Reply:
In fact using `__import__` or `import_module` is just preference, I don't see differences in practices.
But `__import__` have some inconsistency (the same to `import_module`):

`__import__("a.b")` != `__import__("a", fromlist=["b"]).b`

So if you want minimal changes, this way would work:

```diff
    try:
        return getattr(mod, fromlist[0])
    except AttributeError:
+       return __import__(path)
```

I overthought it before.

Reply:
Well I am wrong. Your code is just right. I know so little about `__import__`